thirdparty/DiskANN/include/diskann/utils.h

```diff
@@ -583,6 +585,12 @@ namespace diskann {
   template<typename T>
   float prepare_base_for_inner_products(const std::string in_file,
                                         const std::string out_file) {
+    if (!knowhere::KnowhereFloatTypeCheck<T>::value) {
+      std::stringstream stream;
+      stream << "DiskANN currently only supports floating point data for IP."
```

Comment:
I would interpret this message that only `float` is supported, but not `fp16` or `bf16`. Please mention `fp16` and `bf16` if they are also supported.

Reply:
updated